lib/response-stream.js

```diff
@@ -47,10 +47,15 @@ var ResponseStream = module.exports = function (options) {
   });
 
   if (this.response) {
-    this._headers = this.response._headers = this.response._headers || {};
 
-    // Patch to node core
-    this.response._headerNames = this.response._headerNames || {};
+    try {
+      this._headers = this.response.getHeaders() || {};
+    } catch (err) {
+      this._headers = this.response._headers = this.response._headers || {};
+
+      // Patch to node core
+      this.response._headerNames = this.response._headerNames || {};  
```

Comment:
Shouldn't this be in another try-catch?

```
// Patch to node core
try {
  this.response._headerNames = this.response.getHeaderNames () || {};
} catch (err) {
  this.response._headerNames = this.response._headerNames || {};  
}
```

Reply:
It has been commented in PR #77 that there is no need for this suggestion.
@darrendatbui has did some testing and the original committed code works.

I rest my case.